src/settings-ui/Settings.UI/ViewModels/PowerOcrViewModel.cs

```diff
@@ -130,6 +164,45 @@ public HotkeySettings ActivationShortcut
             }
         }
 
+        private void InitializeLanguages()
+        {
+            possibleOcrLanguages = OcrEngine.AvailableRecognizerLanguages.ToList();
+        }
+
+        internal void SetLanguageSelectedIndex()
+        {
+            int preferredIndex = -1;
+            int systemLanguageIndex = -1;
+            CultureInfo systemCulture = CultureInfo.CurrentUICulture;
+            possibleOcrLanguages = OcrEngine.AvailableRecognizerLanguages.ToList();
+            foreach (Language language in possibleOcrLanguages.OrderBy(x => x.NativeName))
+            {
+                if (_powerOcrSettings.Properties.PreferredLanguage?.Equals(language.DisplayName) == true)
+                {
+                    preferredIndex = AvailableLanguages.Count;
+                }
+
+                if (systemCulture.DisplayName.Equals(language.DisplayName) || systemCulture.Parent.DisplayName.Equals(language.DisplayName))
+                {
+                    systemLanguageIndex = AvailableLanguages.Count;
+                }
+
+                AvailableLanguages.Add(language.NativeName);
+            }
+
+            if (preferredIndex == -1)
+            {
+                preferredIndex = systemLanguageIndex;
+            }
+
+            if (preferredIndex == -1)
+            {
+                preferredIndex = 0;
+            }
```

Comment:
Let's make it easier to read. It's not so clear in the first moment.

Reply:
Thanks for the comment, will simplify the code and also add update of available languages.